File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        no_falso_ancora = ListNode(0, head)

        ponteiro_tras = no_falso_ancora
        ponteiro_frente = no_falso_ancora

        for _ in range (n):
            ponteiro_frente = ponteiro_frente.next

        while ponteiro_frente.next is not None:
            ponteiro_frente = ponteiro_frente.next
            ponteiro_tras = ponteiro_tras.next

        ponteiro_tras.next = ponteiro_tras.next.next

        return no_falso_ancora.next

File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.cs
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        ListNode no_falso_ancora = new ListNode(0, head);

        ListNode ponteiro_tras = no_falso_ancora;
        ListNode ponteiro_frente = no_falso_ancora;

        for(int i = 0; i < n; i++){
            ponteiro_frente = ponteiro_frente.next;
        }

        while(ponteiro_frente.next != null){
            ponteiro_frente = ponteiro_frente.next;
            ponteiro_tras = ponteiro_tras.next;
        }

        ponteiro_tras.next = ponteiro_tras.next.next;

        return no_falso_ancora.next;
    }
}